src/main/java/duke/Task.java

```diff
@@ -0,0 +1,68 @@
+package duke;
+
+public class Task {
+
+    private String description;
+    private boolean isDone;
+
+    /**
+     * Constructor for a Task object.
+     *
+     * @param description Description of the task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Retrieves the correct status to be displayed based on
+     * if the task is done or not.
+     *
+     * @return Status icon.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    public String getDescription() {
+        return this.description;
+    }
+
+    public void flip() {
+        this.isDone = !isDone;
+    }
+
+    /**
+     * Marks the task as done.
+     */
+    public void markDone() {
+        if (!this.isDone) {
+            this.flip();
+        }
+    }
+
+    /**
+     * Marks the task as undone.
+     */
+    public void markUndone() {
+        if (this.isDone) {
+            this.flip();
+        }
+    }
+
+    /**
+     * Outputs the correct string to be saved in save file.
+     * @return String of data to be stored.
```

Comment:
Remember to add a line spacing in your javadoc

Reply:
Fixed